### PR TITLE
Jenkinsfile updates

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -258,7 +258,7 @@ pipeline {
                     fi
 
                     rm -fr out/results
-                    /opt/ort/bin/ort $LOG_LEVEL --stacktrace analyze $ALLOW_DYNAMIC_VERSIONS_OPTION $USE_CLEARLY_DEFINED_CURATIONS_OPTION -f JSON,YAML -i $PROJECT_DIR/source -o out/results/analyzer
+                    /opt/ort/bin/ort $LOG_LEVEL --stacktrace analyze $ALLOW_DYNAMIC_VERSIONS_OPTION $USE_CLEARLY_DEFINED_CURATIONS_OPTION -i $PROJECT_DIR/source -o out/results/analyzer
```
Comment (fviernau) [inline Jenkinsfile:274]: Would it make to do the same for SpdxDocuments in this commit?

**Comment by sschuberth** [comment]
> Reminder, should be merged after #2847.

Actually, this can (and should, in its current state) be merged before #2847. That's because the evaluator is the only build step for which this adds "UNSTABLE" support, and the evaluator already returns status code 2 for unstable. PR #2847 just extends that idea also to the analyzer and scanner CLI tools, but this PR does not support "UNSTABLE" for the analyzer or scanner build steps.

File: Jenkinsfile (hunk 1 above)
Comment (sschuberth) [reply Jenkinsfile:274]: I'm not sure what you mean, could you please clarify?

Comment (sschuberth) [reply Jenkinsfile:274]: As discussed via chat, we prefer to change the default SpdxDocument report format to only be YAML instead, to match the defaults of our CLI result files.

**Pull request merged**